testutils/inject/base.go

```diff
@@ -33,7 +34,10 @@ func NewBase(name string) *Base {
 
 // Name returns the name of the resource.
 func (b *Base) Name() resource.Name {
-	return b.name
+	if b.NameFunc == nil {
```

Comment:
why is this needed, and why can't you use the resource name of the injected base?

Reply:
I am still working on this. It's not yet ready for review yet. I added this because I couldn't get my tests working without it. Now that my tests work, I'll look into this more closely.

Reply:
Done now :) Got rid of this again.